assets/js/blocks/cart-checkout/mini-cart/component-frontend.tsx

```diff
@@ -30,16 +35,42 @@ const MiniCartBlock = ( { isPlaceholderOpen = false } ): JSX.Element => {
 		isPlaceholderOpen
 	);
 
+	useEffect( () => {
+		const openMiniCartAndRefreshData = () => {
+			dispatch( storeKey ).invalidateResolutionForStore();
+			setSkipSlideIn( false );
+			setIsOpen( true );
+		};
+
+		// Make it so we can read jQuery events triggered by WC Core elements.
+		const removeJQueryAddedToCartEvent = translateJQueryEventToNative(
+			'added_to_cart',
+			'wc-blocks_added_to_cart'
+		);
+
+		document.body.addEventListener(
+			'wc-blocks_added_to_cart',
+			openMiniCartAndRefreshData
+		);
+
+		return () => {
+			removeJQueryAddedToCartEvent();
+
+			document.body.removeEventListener(
+				'wc-blocks_added_to_cart',
+				openMiniCartAndRefreshData
+			);
+		};
+	}, [] );
```

Comment:
Can we not react to our own store requests, or is it not in scope for this PR?

Reply:
How would you do that? I based this code on a similar code snippet we have in the Cart block:

https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/b6f577da454e8efcb8f3b9fcfceffe1bbf8cc3c0/assets/js/blocks/cart-checkout/cart/block.js#L35-L73